src/main/java/com/google/devtools/build/lib/syntax/BinaryOperatorExpression.java

```diff
@@ -189,6 +189,18 @@ private static Object evaluate(
         case PIPE:
           return pipe(x, y, env, location);
 
+        case AMPERSAND:
+          return (Integer) x & (Integer) y;
+
+        case CARET:
+          return (Integer) x ^ (Integer) y;
+
+        case GREATER_GREATER:
+          return (Integer) x >> (Integer) y;
+
+        case LESS_LESS:
+          return (Integer) x << (Integer) y;
```

Comment:
This is a subtle one. The Go implementation uses infinite precision integers, so its left and right shifts operate as if on bit strings infinitely extended with sign bits. In contrast, the Java implementation uses signed int32, so this right shift may cause the sign to change. We should at least detect that case and emit an error. We should also clarify in the documentation what the semantics of shifts are if the implementation uses inexact arithmetic.

I think the long term solution here is for the Java implementation to represent Starlark ints exactly using (the smaller of) int32 or bigint, but that's a bigger task.